src/middlewared/middlewared/plugins/interface/netif_linux/routing.py

```diff
@@ -225,3 +233,35 @@ def _op(self, op, route):
             kwargs[key] = value
 
         ip.route(op, **kwargs)
+
+
+class RuleTable:
+
+    @property
+    def rules(self):
+        rules = []
+        tables = {t.table_id: t for t in RoutingTable().routing_tables.values()}
+        for rule in ip.get_rules():
+            attrs = dict(rule['attrs'])
```

Comment:
For most of the IPRoute examples that I've found, checking if the `attrs` key in the `rule` object should be done first.

```
if 'attrs' not in rule:
     continue
 ```
 
 Furthermore, I'm not sure it's necessary to convert the `attrs` attribute to a dict since the `rule` object has built-in methods for getting the necessary items. Examples below.
 
 ```
 table = rule.get_attr('FRA_TABLE')
 if table is None:
     continue
     
prio = rule.get_attr('FRA_PRIORITY')
if prio is None:
    continue

src = rule.get_attr('FRA_SRC')
...
```

Reply:
We have been using `attrs` directly for routes and it always exist so far ? I have found the same to be true for ip rules as well.